src/app/pages/journal/__tests__/journal.page.spec.ts

```diff
@@ -0,0 +1,201 @@
+import { AuthenticationProvider } from '../../../providers/authentication/authentication';
+import { AuthenticationProviderMock } from '../../../providers/authentication/__mocks__/authentication.mock';
+import { SlotSelectorProvider } from '../../../providers/slot-selector/slot-selector';
+import { DateTimeProvider } from '../../../providers/date-time/date-time';
+import { DateTimeProviderMock } from '../../../providers/date-time/__mocks__/date-time.mock';
+import { AppConfigProvider } from '../../../providers/app-config/app-config';
+import { AppConfigProviderMock } from '../../../providers/app-config/__mocks__/app-config.mock';
+import { AppComponent } from '../../../app.component';
+import { StoreModel } from '../../../shared/models/store.model';
+import { LoadingControllerMock } from '../../../../../mock/ionic-mocks/loading-controller.mock';
+import { MockAppComponent } from '../../../__mocks__/app.component.mock';
+import { JournalNavigationComponent } from '../components/journal-navigation/journal-navigation';
+import { selectVersionNumber } from '../../../../store/app-info/app-info.selectors';
+import { DateTime } from '../../../shared/helpers/date-time';
+import * as journalActions from '../../../../store/journal/journal.actions';
+import { MesError } from '../../../shared/models/mes-error.model';
+import { ErrorPage } from '../../error-page/error';
+import { ErrorTypes } from '../../../shared/models/error-message';
+import { MockBackButton } from './back-button.mock';
+import { MockPlatform } from './platform.mock';
+
+describe('JournalPage', () => {
+  let component: JournalPage;
+  let fixture: ComponentFixture<JournalPage>;
+  let mockBackButton; let
+    mockPlatform;
+  const platformReadySpy = jasmine.createSpy().and.returnValue(Promise.resolve());
+  let store$: MockStore;
+  const initialState = {
+    appInfo: {
+      versionNumber: '4.0',
+    },
+  } as StoreModel;
+  const mockNavParams = {
+    get: (param: string) => {
+      const data = {};
+      return data[param];
+    },
+  };
+
+  beforeEach(async(() => {
+    mockBackButton = new MockBackButton();
+    mockBackButton.subscribeWithPriority = jasmine.createSpy('subscribeWithPriority', () => {
+    });
+    mockPlatform = new MockPlatform();
+    mockPlatform.backButton = mockBackButton;
+    mockPlatform.ready = platformReadySpy;
+
+    const modalSpy = jasmine.createSpyObj('Modal', ['present']);
+    const modalControllerMock = jasmine.createSpyObj('ModalController', ['create']);
+    modalControllerMock.create.and.callFake(() => {
+      return modalSpy;
+    });
+
+    TestBed.configureTestingModule({
+      declarations: [
+        JournalPage,
+        MockComponent(JournalNavigationComponent),
+      ],
+      imports: [
+        IonicModule,
+        RouterTestingModule.withRoutes(
+          [
+            { path: '', component: JournalPage },
+          ],
+        ),
+      ],
+      providers: [
+        { provide: Platform, useValue: mockPlatform },
```

Comment:
can we use the PlatformMock instead of a local mock?

Reply:
im sorry matt, no we cant